core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregatorUtils.java

```diff
@@ -48,8 +50,7 @@ public static DeduplicateAggregationDefinition generateAggregationDefinition(Rel
     String inputRelationName = ctx.getInputRelationNames().stream().findFirst().orElse(null);
     Schema inputSchema = inputRelationName != null ? ctx.getInputSchema(inputRelationName) : null;
 
-    Optional<ExpressionFactory<String>> expressionFactory = ctx.getEngine().
-      getExpressionFactory(StringExpressionFactoryType.SQL);
+    Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(ctx, inputSchema);
```

Comment:
Please handle case when this Optional is empty

Reply:
Good catch, I thought I had added this.